src/plan_runner/state_running.cc

```diff
@@ -0,0 +1,88 @@
+#include "state_idle.h"
+#include "state_running.h"
+#include "state_error.h"
+
+using std::cout;
+using std::endl;
+
+PlanManagerStateBase* StateRunning::instance_ = nullptr;
+PlanManagerStateBase* StateRunning::Instance() {
+  if (!instance_) {
+    instance_ = new StateRunning();
+  }
+  return instance_;
+}
+
+const PlanBase * StateRunning::GetCurrentPlan(PlanManagerStateMachine *state_machine,
```

Comment:
I understand the logic here (this is being called every 200Hz on the main thread), but the method name here a bit confusing, since it's doing a lot more than `GetCurrentPlan`, and is really in charge of starting, running, and stopping the plan (especially when in running state)

How about something like `HandleCurrentPlan`?

Reply:
I prefer to reserve names starting with Handle for lcm callbacks. What about `GetOrDiscardCurrentPlan`?

Reply:
I think the right thing to do is to separate out the part that is starting and stopping the plan into a different method, to avoid it being overloaded with a method that returns the plan. 

But we can just merge and fix later :) I think it is a nit.